packages/mdc-textfield/index.ts

```diff
@@ -409,21 +349,21 @@ class MDCTextField extends MDCComponent {
     this.foundation_.notchOutline(openNotch);
   }
 
-  /**
-   * @return {!MDCTextFieldFoundation}
-   */
-  getDefaultFoundation() {
-    return new MDCTextFieldFoundation(
-      /** @type {!MDCTextFieldAdapter} */ (Object.assign({
+  getDefaultFoundation(): MDCTextFieldFoundation {
+    return new MDCTextFieldFoundation({
+      ...({
+        // tslint:disable:object-literal-sort-keys
         addClass: (className) => this.root_.classList.add(className),
         removeClass: (className) => this.root_.classList.remove(className),
         hasClass: (className) => this.root_.classList.contains(className),
         registerTextFieldInteractionHandler: (evtType, handler) => this.root_.addEventListener(evtType, handler),
         deregisterTextFieldInteractionHandler: (evtType, handler) => this.root_.removeEventListener(evtType, handler),
         registerValidationAttributeChangeHandler: (handler) => {
-          const getAttributesList = (mutationsList) => mutationsList.map((mutation) => mutation.attributeName);
+          const getAttributesList = (mutationsList: MutationRecord[]) => {
+            return mutationsList.map((mutation) => mutation.attributeName!).filter((attributeName) => attributeName);
```

Comment:
Why was this changed?

Reply:
`handler()` expects an argument of type `string[]`, but `mutation.attributeName` can technically return `null`. Without `filter()`, `getAttributesList()` returns `(string | null)[]`.